app/js/lib/foxbox/services.js

```diff
@@ -0,0 +1,246 @@
+import EventDispatcher from './event-dispatcher';
+import BaseService from './services/base';
+import IpCameraService from './services/ip-camera';
+import LightService from './services/light';
+import DoorLockService from './services/door-lock';
+import MotionSensorService from './services/motion-sensor';
+
+const p = Object.freeze({
+  api: Symbol('api'),
+  settings: Symbol('settings'),
+  db: Symbol('db'),
+  cache: Symbol('services'),
+  isPollingEnabled: Symbol('isPollingEnabled'),
+  nextPollTimeout: Symbol('nextPollTimeout'),
+
+  // Private methods.
+  schedulePoll: Symbol('schedulePoll'),
+  getServiceInstance: Symbol('getServiceInstance'),
+  hasDoorLockChannel: Symbol('hasDoorLockChannel'),
+  updateServiceList: Symbol('updateServiceList'),
+  getCache: Symbol('getCache'),
+});
+
+/**
+ * Compare 2 objects. Returns true if all properties of object A have the same
+ * value in object B. Extraneous properties in object B are ignored.
+ * Properties order is not important.
+ *
+ * @param {Object} objectA
+ * @param {Object} objectB
+ * @return {boolean}
+ */
+const isSimilar = (objectA, objectB) => {
+  const keysA = Object.keys(objectA);
+  const keysB = Object.keys(objectB);
+
+  if (keysA.length !== keysB.length) {
+    return false;
+  }
+
+  return !keysA.some((keyA) => {
+    const valueA = objectA[keyA];
+    const valueB = objectB[keyA];
+
+    const typeA = typeof valueA;
+
+    if (typeA !== typeof valueB) {
+      return true;
+    }
+
+    if (typeA !== 'object' || valueA === null || valueB === null) {
+      return valueA !== valueB;
+    }
+
+    return !isSimilar(valueA, valueB);
+  });
+};
+
+export default class Services extends EventDispatcher {
```

Comment:
Ah yeah I forgot about `foxbox.recipes`. I need to give it some thoughts, don't change anything for now. This approach is probably the best one.